drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1115,6 +1122,8 @@ static void rx_timeout(struct k_timer *timer)
 
 }
 
+#ifdef UARTE_ANY_ASYNC
```

Comment:
Instead of this, `CONFIG_UART_ASYNC_API` should be replaced with `UARTE_ANY_ASYNC` in this file. There is no point in keeping stuff related to the asynchronous API when it has no chance of being used.

Reply:
Done